scripts/Muon/GUI/Common/plot_widget/plot_widget_presenter.py

```diff
@@ -11,16 +11,21 @@
 from Muon.GUI.Common.plot_widget.external_plotting.external_plotting_view import ExternalPlottingView
 from Muon.GUI.Common.plot_widget.plotting_canvas.plotting_canvas_presenter_interface import \
     PlottingCanvasPresenterInterface
+from Muon.GUI.Common.contexts.frequency_domain_analysis_context import FrequencyDomainAnalysisContext
 from Muon.GUI.Common.plot_widget.plot_widget_model import PlotWidgetModel
 from Muon.GUI.Common.plot_widget.plot_widget_view_interface import PlotWidgetViewInterface
+from Muon.GUI.Common.contexts.muon_gui_context import PlotMode
 from mantidqt.utils.observer_pattern import GenericObserver, GenericObserverWithArgPassing, GenericObservable
 from mantid.dataobjects import Workspace2D
 
+MUON_ANALYSIS_DEFAULT_X_RANGE = [0.0, 15.0]
+FREQUENCY_DOMAIN_ANALYSIS_DEFAULT_X_RANGE = [0.0, 1000.0]
```

Comment:
should we get these from the context?

Reply:
Yes good idea that removes the need for one of the if statements.